src/main/java/com/pnu/dev/radioserviceapi/client/YoutubeApiClientImpl.java

```diff
@@ -0,0 +1,83 @@
+package com.pnu.dev.radioserviceapi.client;
+
+import com.pnu.dev.radioserviceapi.client.dto.search.ItemYoutubeSearchResponse;
+import com.pnu.dev.radioserviceapi.client.dto.search.YoutubeSearchResponse;
+import com.pnu.dev.radioserviceapi.client.dto.videos.ItemYoutubeVideosResponse;
+import com.pnu.dev.radioserviceapi.client.dto.videos.YoutubeVideosResponse;
+import com.pnu.dev.radioserviceapi.exception.ServiceException;
+import lombok.Data;
+import org.apache.http.client.utils.URIBuilder;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URISyntaxException;
+import java.util.List;
+
+
+@Component
+@Data
+public class YoutubeApiClientImpl implements YoutubeApiClient {
+
+    private static final String GET_VIDEOS_BY_CHANNEL_URI =
+            "https://www.googleapis.com/youtube/v3/search?part=snippet&order=date&type=video";
+
+    private static final String GET_VIDEO_BY_ID_URI =
+            "https://www.googleapis.com/youtube/v3/videos?part=snippet";
+
+
+    @Value("${youtube.api_key}")
+    private String api_key;
+
+    @Value("${youtube.channel_id}")
+    private String CHANNEL_ID;
+
+    @Value("${youtube.max_results_number}")
+    private int MAX_RESULTS_NUMBER;
+
+    private final RestTemplate restTemplate;
+
+    public YoutubeApiClientImpl() {
+        this.restTemplate = new RestTemplate();
+    }
+
+    @Override
+    public List<ItemYoutubeSearchResponse> getLastVideos() {
+
+        URIBuilder uriRequest;
+        try {
+            uriRequest = new URIBuilder(GET_VIDEOS_BY_CHANNEL_URI);
+        } catch (URISyntaxException e) {
+            throw new ServiceException("Bad uri Youtube.api");
+        }
+        uriRequest.addParameter("key", api_key)
+                .addParameter("channelId", CHANNEL_ID)
+                .addParameter("maxResults", String.valueOf(MAX_RESULTS_NUMBER));
+        YoutubeSearchResponse responseEntity = restTemplate.getForObject(uriRequest.toString(), YoutubeSearchResponse.class);
+        if (responseEntity != null) {
+            return responseEntity.getItems();
+        } else {
+            throw new ServiceException("Youtube api response error");
```

Comment:
As long as you use this class for handling api requests as well as requests from admin panel, don't throw such exceptions from here.

Use something like this as a return type and throw appropriate exception on service layer:
```

@Data
@Builder(access = AccessLevel.PRIVATE)
@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
class OperationResult<T> {

    T data;

    String errorMessage;

    public boolean isError() {
        return StringUtils.isNotBlank(errorMessage);
    }

    public static <T> OperationResult<T> success(T data) {
        return OperationResult.<T>builder()
                .data(data)
                .build();
    }

    public static <T> OperationResult<T> error(String errorMessage) {
        return OperationResult.<T>builder()
                .errorMessage(errorMessage)
                .build();
    }

}
```